### ISSUE TITLE
Wrong return type when autocomplete async abstractmethod

### ISSUE THREAD

**Issue by E5presso** [microsoft/pylance-release#4573]
## Environment data

-   Language Server version: v2023.5.30
-   OS and version: OS X Ventura (13.4.1)
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.11.3 & using Poetry 1.4.2

## Code Snippet

```python
from abc import ABC, abstractmethod


class Abstract(ABC):
    @abstractmethod
    async def do_async(self) -> None:
        ...


class Implement(Abstract):
    ...
```

## Repro Steps

1. When I type 'async def do_' in class definition to trigger autocomplete suggestion.
2. Pylance suggests to autocomplete method declaration.
3. Press Enter to apply autocompletion.

## Expected behavior

The autocompletion of abstract async method should be
```python
class Implement(Abstract):
    async def do_async(self) -> None:
        return await super().do_async()
```

## Actual behavior

But the autocompletion printed like
```python
class Implement(Abstract):
    async def do_async(self) -> Coroutine[Any, Any, None]:
        return await super().do_async()
```

## Logs

```
Nothing printed out
```

**Comment by heejaechang** [comment]
another issue to take a look. for this

```python
async def asyncGen():
    await asyncio.sleep(1)
    yield 1
```

we show return type as `Generator` rather than `AsyncGenerator` but when `asyncGen` is used in `async for ...`, it shows its return type as `AsyncGenerator`

also, when we generate `overriden` method that returns `AsyncGenerator`, we shouldn't put `await` before `super().xxxx`

**Comment by erictraut** [comment]
> we show return type as `Generator` rather than `AsyncGenerator`

This was fixed in last week's pyright. See [this issue](https://github.com/microsoft/pyright/issues/5411) for details.

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2023.7.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023731-19-July-2023-PreRelease)

**Comment by rchiodo** [comment]
In case it's not obvious, you'll need to type `async def do_` in order for the completion to come out how you described above. If you type just `def do_`, it will complete without async so the return will be a Coroutine.
